makeflow/src/makeflow_module_sandbox.c

```diff
@@ -0,0 +1,90 @@
+/*
+ Copyright (C) 2018- The University of Notre Dame
+ This software is distributed under the GNU General Public License.
+ See the file COPYING for details.
+ */
+
+#include "dag_file.h"
+#include "makeflow_log.h"
+#include "makeflow_hook.h"
+
+#include "batch_task.h"
+#include "batch_wrapper.h"
+
+#include "debug.h"
+#include "list.h"
+#include "stringtools.h"
+#include "xxmalloc.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+static int makeflow_module_sandbox_node_submit(struct dag_node *node, struct batch_task *task){
+	struct batch_wrapper *wrapper = batch_wrapper_create();
+	char *wrap_name = string_format("task_%d_sandbox", task->taskid);
+	batch_wrapper_prefix(wrapper, wrap_name);
+
+	/* Save the directory we were originally working in. */
+	batch_wrapper_pre(wrapper, "export CUR_WORK_DIR=$(pwd)");
+
+	/* Create sandbox. This should probably have a hex or random tail to be unique. */
+	char *cmd = string_format("mkdir %s", wrap_name);
```

Comment:
This is probably fine, but it wouldn't be too hard to change this line to

    char *cmd = string_format("export SANDBOX=$(mktemp -d %s_XXXXXX)", wrap_name);

Then subsequent commands could just refer to `$SANDBOX`.